syntax = "proto3";

message LoRaPacket {
    fixed32 hardwareID = 1;

    oneof payload {
        RoverData roverData = 2;
        RoverDiscovery roverDiscovery = 3;
        RoverConfiguration roverConfiguration = 4;
    }
}

message RoverData {
    float latitude = 1;
    float longitude = 2;
    uint32 heading = 3;
    int32 heel = 4;
}

message RoverDiscovery {
}

message RoverConfiguration {
    repeated int32 slots = 1;
}
